src/vs/workbench/contrib/debug/browser/variablesView.ts

```diff
@@ -53,18 +55,28 @@ export class VariablesView extends ViewletPanel {
 
 		// Use scheduler to prevent unnecessary flashing
 		this.onFocusStackFrameScheduler = new RunOnceScheduler(() => {
+			const stackFrame = this.debugService.getViewModel().focusedStackFrame;
+
 			this.needsRefresh = false;
-			this.tree.updateChildren().then(() => {
-				const stackFrame = this.debugService.getViewModel().focusedStackFrame;
-				if (stackFrame) {
-					stackFrame.getScopes().then(scopes => {
-						// Expand the first scope if it is not expensive and if there is no expansion state (all are collapsed)
-						if (scopes.every(s => this.tree.getNode(s).collapsed) && scopes.length > 0 && !scopes[0].expensive) {
-							this.tree.expand(scopes[0]).then(undefined, onUnexpectedError);
-						}
-					});
+			if (stackFrame && this.savedViewState) {
```

Comment:
I think this whole logic can be simplified.
I would remove the checks for the `this.savedViewState` and simply
`if (stackFrame)` set the input pass this.savedViewState (even if undefined it should be fine)
`else`
set `savedViewState` and do `updateChildren()`

Reply:
At [line 65](https://github.com/haneefdm/vscode/blob/003fb37af99fa13d449c83c6de34d6540699df61/src/vs/workbench/contrib/debug/browser/variablesView.ts#L65), the check can certainly be simplified to `if (!stackFrame) {`, but removing the check competely (always saving) has a consequence along with the change you suggest at  [line 61,](https://github.com/haneefdm/vscode/blob/003fb37af99fa13d449c83c6de34d6540699df61/src/vs/workbench/contrib/debug/browser/variablesView.ts#L61)

With your suggested changes code would look like
```javascript
this.needsRefresh = false;
if (stackFrame) {
    this.tree.setInput(this.debugService.getViewModel(), this.savedViewState).then(null, onUnexpectedError);
    this.savedViewState = undefined;
} else {
    this.savedViewState = this.tree.getViewState();
    this.tree.updateChildren().then(null, onUnexpectedError);
}
```
Currently, we have a nice feature of automatically expanding the first node in the tree. If I simplify it to just `if (stackFrame) {` at the start, then we lose that feature. and then I have to remove lines 70-77 resulting in the code you see above. It affects the start of a session and when focus changes to a different frame. I think people will complain.

I saw that the scheduler is not only used when the focus changes in the call-stack but also when the panels visibility changes, so I was being paranoid and not overwrite a good saved state to a potentially empty state. I don't think it can happen so I will simplify line 65.

Reply:
I tested and pushed the change for a simplified check at line 65.

Please let me know if you want any other change. I will only be glad and will work on it until you are satisfied.